src/classes/PMT_Payment_TDTM.cls

```diff
@@ -313,6 +316,52 @@ public class PMT_Payment_TDTM extends TDTM_Runnable {
         }
     }
 
+    /*******************************************************************************************************
+    * @description Validate that the opportunity has the appropriate values. This will check to make sure
+    * that the opportunity amount is not set to an amount lower than the paid payments.
+    * @param newlist The list of Opportunities from trigger new.
+    * @param oldlist The list of Opportunities from trigger old.
+    * @param triggerAction  Which trigger event (BeforeInsert, AfterInsert, etc.).
+    * @return void Does not return anything
+    ********************************************************************************************************/
+    private void validateOpportunity(Map<Id, Opportunity> newMap, Map<Id, Opportunity> oldMap, TDTM_Runnable.Action triggerAction, DmlWrapper dmlWrapper) {
+        if (triggerAction == TDTM_Runnable.Action.BeforeUpdate) {
+            enforceFinalizedOpportunityAmountIsEqualToPayment(newMap, oldMap);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Makes sure that for finalized opportunities, the amount is equal to the paid payment amount
+    * If the user tries to change the amount, throw an error telling them to change the payments instead.
+    * @param opportunitiesToCheck The list of opportunities to check the amount for
+    * @return void Does not return anything.
+    ********************************************************************************************************/
+    private void enforceFinalizedOpportunityAmountIsEqualToPayment(Map<Id, Opportunity> newMap, Map<Id, Opportunity> oldMap) {
+
+        // Get the opportunities with payments.
+        Map<Id, Opportunity> oppsById = getPaymentOpportunities(newMap.keySet());
+
+        for (Opportunity queriedOpp : oppsById.values()) {
+            Opportunity oldOpp = oldMap.get(queriedOpp.Id);
+            Opportunity updatedOpp = newMap.get(queriedOpp.Id);
+            if (oldOpp.Amount != updatedOpp.Amount && OPP_StageMappingUtil.determineOppStateFromOppStage(updatedOpp.StageName) == OPP_StageMappingUtil.getFinalizedState()) {
```

Comment:
Would be better if it passed in the stage to a method in that class that returned true or false, as we will do this in other spots...
Added to OPP_StageMappingUtil: 
public static Boolean isFinalized(String stageName) {
   return OPP_StageMappingUtil.determineOppStateFromOppStage(stageName) == FINALIZED_STATE_VAL;
}

Then it becomes (if (oldOpp.Amount != updatedOpp.Amount && OPP_StageMappingUtil.isFinalized(updateOpp.StageName))

Or you could even pass the Opportunity, and get the Opportunity stage inside the method.